corehq/apps/userreports/views.py

```diff
@@ -215,8 +216,26 @@ def page_context(self):
         return {
             'form': self.edit_form,
             'report': self.config,
+            'referring_apps': self.get_referring_apps(),
         }
 
+    def get_referring_apps(self):
+        to_ret = []
+        apps = get_apps_in_domain(self.domain)
```

Comment:
It took 4 or 5 seconds to run this `get_referring_apps` for a random report on ICDS. For a typical domain with around 10 apps, even if they are as large as ICDS it shouldn't take much more than this, right?

It would be definitely slow, if were to iterate through all builds.

Reply:
Did you check on icds-cas on india or TCL? The india domain will be slower because it has other test apps in the domain.

4 to 5 seconds seems reasonable for this. Not sure if that will start stretching into ~1 min range on india (probably ok since as you say these aren't super often accessed